syntax = "proto3";
package cdh.telemetry;

import "subsystems.proto";

message Telemetry {
  cdh.subsystems.Subsystem sys = 1;
  uint32 ID = 2;
  oneof data {
    bool bool_value = 3;
    double double_value = 4;
    int32 int_value = 5;
    bytes bytes_value = 6;
  }
  uint32 timestamp = 7; // Computation timestamp, in case collection for a full frame is long
  uint32 checksum = 8;
}

message TMFrame {
  uint32 timestamp = 1; // Sending timestamp
  uint32 sequence_no = 2;
  repeated Telemetry TM = 3;
}
